pallets/staking/src/pallet/mod.rs

```diff
@@ -0,0 +1,2113 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Staking FRAME Pallet.
+
+use frame_support::{
+    dispatch::Codec,
+    pallet_prelude::*,
+    traits::{
+        Currency, CurrencyToVote, EnsureOrigin, EstimateNextNewSession, Get, LockableCurrency,
+        OnUnbalanced, UnixTime,
+    },
+    weights::Weight,
+    BoundedVec,
+};
+use frame_system::{ensure_root, ensure_signed, pallet_prelude::*};
+use sp_runtime::{
+    traits::{CheckedSub, SaturatedConversion, StaticLookup, Zero},
+    ArithmeticError, Perbill, Percent,
+};
+use sp_staking::{EraIndex, SessionIndex};
+use sp_std::prelude::*;
+
+mod impls;
+
+use crate::{
+    slashing, weights::WeightInfo, AccountIdLookupOf, ActiveEraInfo, BalanceOf, EraRewardPoints,
+    Exposure, Forcing, NegativeImbalanceOf, Nominations, PositiveImbalanceOf, RewardDestination,
+    SessionInterface, StakingLedger, UnappliedSlash, ValidatorPrefs,
+};
+
+use frame_support::dispatch::DispatchClass::Operational;
+use frame_support::traits::schedule::Anon;
+use frame_support::traits::IsSubType;
+use frame_support::weights::constants::{WEIGHT_REF_TIME_PER_MICROS, WEIGHT_REF_TIME_PER_NANOS};
+use frame_system::offchain::SendTransactionTypes;
+use sp_npos_elections::ElectionScore;
+use sp_runtime::curve::PiecewiseLinear;
+use sp_runtime::traits::{Dispatchable, Saturating};
+use sp_runtime::Permill;
+
+use polymesh_common_utilities::identity::Config as IdentityConfig;
+use polymesh_common_utilities::{Context, GC_DID};
+use polymesh_primitives::IdentityId;
+
+use crate::types::{
+    ElectionCompute, ElectionResult, ElectionSize, ElectionStatus, PermissionedIdentityPrefs,
+    SlashingSwitch,
+};
+use crate::{CompactAssignments, ValidatorIndex, MAX_ALLOWED_VALIDATORS, STAKING_ID};
+
+type Identity<T> = pallet_identity::Module<T>;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+    /// The current storage version.
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(7);
```

Comment:
We use to control migrations using our own type `Releases`.  Instead of using the macro should we implement this manually?

Reply:
I think the `Releases` type was how the Substrate code was doing migrations.  We added `PolymeshStorageVersion` to avoid messing with the storage version used by the Substrate code.

Might be best to keep using our `PolymeshStorageVersion` to handle migrations of our own storage items.